src/btree/bt_delete.c

```diff
@@ -164,6 +164,16 @@ __wt_delete_page(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
         goto err;
     if (addr.ta.prepare)
         goto err;
+
+    /*
+     * When performing a truncate operation with no associated timestamp, limit fast-truncate to
+     * pages where all its data is globally visible.
+     */
+    if (F_ISSET(session->txn, WT_TXN_TS_NOT_SET) &&
+      !__wt_txn_visible_all(session, addr.ta.newest_txn,
+        WT_MAX(addr.ta.newest_start_durable_ts, addr.ta.newest_stop_durable_ts)))
```

Comment:
I think we only need to check the start durable ts. Even the stop durable ts is not globally visible, it cannot be moved to the history store and should be fine.

Reply:
If there is a stop timestamp on the key and it is visible to the truncate operation, the truncate operation just skips the key. So there is no problem with the pages that have keys that have already been removed.

@mm-ng Can you add a comment here? Something like `checking with/without `newest_stop_durable_ts` doesn't have any problem, but for consistency, we check for the maximum of both start and stop timestamps`.